File: java/GroupAnagrams.java
public class GroupAnagrams {
    public List<List<String>> groupAnagrams(String[] strs) {
        public List<List<String>> groupAnagrams(String[] strs) {
            HashMap<String, List<String>> map = new HashMap<>();
            for (String s:strs){
                int[] charArray = new int[26];
                for (char c: s.toCharArray()){
                    charArray[c -'a']++;
                }
                String key = Arrays.toString(charArray);
                map.putIfAbsent(key,new ArrayList<>());
                map.get(key).add(s);
            }
            return new ArrayList<>(map.values());
        }
    }
}

File: python/Group_Anagrams.py
from typing import List


def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
    res = {}
    for string in strs:
        key = [0]*26
        for c in string:
            key[ord(c)-ord('a')]+=1
        if str(key) in res:
            res[str(key)].append(string)
        else:
            res[str(key)] = [string]
    return res.values()